### ISSUE TITLE
Create proven headers rules and add to POS Network class + Test

### ISSUE THREAD

**Issue by noescape00** [stratisproject/StratisBitcoinFullNode#1791]
Also consider issue #979.

**Comment by dangershony** [comment]
For this task we have to be careful not to mix PH in to the low level code (for example PH are not needed for Bitcoin)  This means PH should be part of rules or an override of one of the validators.

**Comment by noescape00** [comment]
See for additional info: https://github.com/stratisproject/StratisBitcoinFullNode/issues/1792

**Comment by noescape00** [comment]
Depends on: https://github.com/stratisproject/StratisBitcoinFullNode/issues/1790
